### DISCUSSION TITLE
Is there a known issue with the profanity filter? Some of the most well known expletives are not being filtered

### DISCUSSION THREAD

**Discussion by rilhia** [deepgram/community#101] [General help] [answered]
I have been playing with this for a couple of days and think it is awesome! I see so many possibilities for putting this to use. While playing, I have been trying out as much as possible. My latest piece involves using the API (via Java) to stream from my microphone. My other half and I were playing around and I mentioned the profanity filter. So we tried it out and had a conversation with some of the most obscene language we could get in. Yes, very childish, but we all are at times ;-) The thing is, some well known profane words were returned without any filtering. For example, a word beginning with F was returned every time. As was a word beginning with S. There was also a word beginning with C which was also regularly returned....when it wasn't misunderstood for "can't" ....please don't judge me  😉

The websocket URI I am using can be seen below....

"wss://api.deepgram.com/v1/listen?language=en&tier=enhanced&model=meeting&diarize=true&smart_format=true&encoding=linear16&sample_rate=16000&profanity_filter=true"

The streaming websocket functionality is great! I am tempted to put something together for taking minutes for meetings. We originally were testing whether the websocket functionality would be able to spot multiple people if they weren't talking at the same time. It did. We were also carrying out an experiment with calculating the time consumed in a discussion by each user. My other half then asked about the possibility of assessing the tone of people's language in a conversation. I said that it wasn't built-in, but I could build something in to search for particular words quite easily. Then I remembered the profanity filter and we tested it. When I saw the results, I thought that I should raise it.

I haven't tried this with prerecorded content and this was just a silly little experiment to see how well it worked. But this didn't seem to work.

**Comment by shirgoldbird** [comment, marked as answer]
@rilhia Thanks for flagging this! Unfortunately, our profanity filter does not currently work for enhanced or Nova-tier models—only the base tier. I've put in an update to the documentation to clarify this.

We should have some updates to the profanity filter functionality and compatibility coming out in the next few months. If you'd be interested in beta testing, please let me know!

**Reply by rilhia** [reply]
Hi @shirgoldbird,

Thanks for getting back to me about this. It wasn't a massive issue for me, but thought I should raise it. I am not easily offended :-)

But I will be more than happy to test out any improvements.
